### ISSUE TITLE
[Feature]: Ability to intercept and abort long running requests based on response size

### ISSUE THREAD

**Issue by heaven** [puppeteer/puppeteer#11523]
### Feature description

While scraping web pages sometimes they try to load huge resources, like 20-30 MB images. Which often leads to chromium crashes, OOM errors, or just timeouts.

It would be ideal if there was a way to set the limit, e.g. a few megabytes and once this threshold is exceeded the request would be aborted, raise an exception, or even give us the ability to capture the URL of a failed request so we can report that to the website owner as a recommendation to fix.

Probably I'm just lost in the docs and someone could help me by pointing to the existing solution but so far I wasn't able to find anything relevant, except for #1191.

**Comment by OrKoN** [comment]
There are params in CDP that Puppeteer currently does not configure https://chromedevtools.github.io/devtools-protocol/tot/Network/#method-enable but there should be some sensible default limits on the Puppeteer level. There is nothing to intercept responses and it might not always work since responses might be streaming and, thus, of unknown size at the time of interception.

**Comment by heaven** [comment]
Yeah, I feel that interception is not something that would be suitable for this. Rather a property on page/browser level that would just abrupt those requests that exceed a given threshold.
